analyse.py

```diff
@@ -298,12 +296,196 @@ def get_file_messages(fp):
 
 
 def get_folder_messages(input_dir):
-    files = [f for f in os.listdir(input_dir) if f.endswith(".json")]
+    files = [f for f in os.listdir(input_dir) if f.endswith(".json") and f.startswith("message_")]
     messages = []
     for f in files:
         messages.extend(get_file_messages(os.path.join(input_dir, f)))
     return messages
 
+def get_participants(input_dir):
+    files = [f for f in os.listdir(input_dir) if f.endswith(".json") and f.startswith("message_")]
+    participants = set()
+    for f in files:
+        print(os.path.join(input_dir, f))
+        with open(os.path.join(input_dir, f), "r") as f:
+            data = json.load(f)
+            for participant in data["participants"]:
+                participants.add(participant["name"])
+    return participants
+
+
+def save_nicknames(messages, input_dir, OUTDIR):

```

Comment:
would prefer to have `OUTDIR` in lower case here as its an argument and not a constant